### PR TITLE
mqtt plugin

### PR DESCRIPTION
**Pull request by angelopoerio**
MQTT is a very popular protocol in the iot/embedded devices world ... so a plugin for it would be nice

### CODE REVIEW

File: src/plugins/mqtt/mod.rs [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+
+        let cli = mqtt::AsyncClient::new(create_opts).map_err(|e| e.to_string())?;
+
+        let conn_opts = mqtt::ConnectOptionsBuilder::new()
+            .connect_timeout(timeout)
+            .user_name(creds.username.to_owned())
+            .password(creds.password.to_owned())
+            .finalize();
+
+        cli.connect(conn_opts).await.map_err(|e| e.to_string())?;
```
Comment (evilsocket) [inline src/plugins/mqtt/mod.rs:61]: there's a problem here, for a failed login attempt, None should be returned ... any error is considered either a timeout or something that would trigger a retry

Comment (angelopoerio) [reply src/plugins/mqtt/mod.rs:61]: Will address this tomorrow

Comment (evilsocket) [reply src/plugins/mqtt/mod.rs:61]: no rush! and thank you so much

Comment (angelopoerio) [reply src/plugins/mqtt/mod.rs]: @evilsocket addressed your comment and added support for mqtt v5 (default is v3.x)

**Comment by BitThr3at** [comment]
how can this plugin connect to mqtt over TLS? it fails with default command.

**Pull request merged**